### ISSUE TITLE
Problems with variant_impacts.transcript field

### ISSUE THREAD

**Issue by naumenko-sa** [arq5x/gemini#798]
Good morning!

Thanks for the excellent framework.

I wanted to extract all of the impacts for the mutations, including all of the transcripts affected.

It tried the query:
```
select v.chrom, v.start+1,v.end, v.ref, v.alt, v.gene,vi.gene, v.transcript, vi.transcript,
                             vi.aa_change,vi.aa_length 
                             from variant_impacts vi, variants v 
                             where vi.variant_id=v.variant_id and  v.is_coding = 1
```
And it seems that variant_impacts.transcript has the same value for all of the transcripts of a particular gene - it is the value of
variants.transcript. However aa_change and aa_length are different for different transcript, as it should be.

How to fix the transcript names, or what am I doing wrong?

Sergey

**Comment by brentp** [comment]
Hi Sergey, what version of gemini are you using? Can you show some output of your query? When running this on test.query.db from the gemini test-suite, I see, e.g.:
```
chr1|69270|69270|A|G|OR4F5|OR4F5|ENST00000335137|ENST00000335137|S60|305
chr1|69428|69428|T|G|OR4F5|OR4F5|ENST00000335137|ENST00000335137|F113C|305
chr1|69511|69511|A|G|OR4F5|OR4F5|ENST00000335137|ENST00000335137|T141A|305
chr1|69761|69761|A|T|OR4F5|OR4F5|ENST00000335137|ENST00000335137|D224V|305
chr1|69871|69871|G|A|OR4F5|OR4F5|ENST00000335137|ENST00000335137|D261N|305
chr1|69897|69897|T|C|OR4F5|OR4F5|ENST00000335137|ENST00000335137|S269|305
```

where all transcripts match, likely because there is only 1. at that location, but later, I see:
```
chr1|874665|874665|G|A|SAMD11|SAMD11|ENST00000420190|ENST00000420190|L177|178
chr1|874665|874665|G|A|SAMD11|SAMD11|ENST00000420190|ENST00000342066|L177|681
chr1|874665|874665|G|A|SAMD11|SAMD11|ENST00000420190|ENST00000478729||
chr1|874665|874665|G|A|SAMD11|SAMD11|ENST00000420190|ENST00000464948||
chr1|874665|874665|G|A|SAMD11|SAMD11|ENST00000420190|ENST00000466827||
chr1|874665|874665|G|A|SAMD11|SAMD11|ENST00000420190|ENST00000474461||
```

where nearly all transcripts are different.

**Comment by naumenko-sa** [comment]
Hi Brent!

I'm using gemini 0.19.1 installed by bcbio.
```
gemini --version
gemini 0.19.1
```

I see just 1 impact per gene (all impacts, coding and non-coding):
```
select count(*) from variant_impacts;
144942
```
```
chr1	860521	860521	C	A	SAMD11	SAMD11	ENST00000342066	ENST00000342066	-/681
chr1	861630	861630	G	A	SAMD11	SAMD11	ENST00000342066	ENST00000342066	-/681
chr1	866319	866319	G	A	SAMD11	SAMD11	ENST00000342066	ENST00000342066	-/681
chr1	866511	866511	C	CCCCT	SAMD11	SAMD11	ENST00000342066	ENST00000342066	-/681
chr1	871334	871334	G	T	SAMD11	SAMD11	ENST00000342066	ENST00000342066	-/681
chr1	874950	874950	T	TCCCTGGAGGACC	SAMD11	SAMD11	ENST00000342066	ENST00000342066	-/681
chr1	876499	876499	A	G	SAMD11	SAMD11	ENST00000342066	ENST00000342066	-/681
```
It is when I'm using VEP annotation with HGVSc,HGVSp.
When using SNPeff I see many impacts for a gene:
```
select count(*) from variant_impacts;
737809
``` 

```
chr1	866319	866319	G	A	SAMD11	SAMD11	ENST00000437963	ENST00000437963	-/109
chr1	866319	866319	G	A	SAMD11	SAMD11	ENST00000437963	ENST00000342066	-/681
chr1	866319	866319	G	A	SAMD11	SAMD11	ENST00000437963	ENST00000341065	-/589
chr1	866319	866319	G	A	SAMD11	SAMD11	ENST00000437963	ENST00000420190	-/179
```

I need both VEP and impacts for all transcripts.
Thanks,
Sergey

**Comment by brentp** [comment]
I'm having trouble understanding the problem. Can you share a small VCF that I can use to recreate?

**Comment by naumenko-sa** [comment]
Sure,
https://drive.google.com/open?id=0B_bLL10GwDnsem9BS2tIY2tvbW8
Thanks,
Sergey

**Comment by brentp** [comment]
that VCF was annotated with snpEff, not VEP. I suspect you'll have a better result if you specify -t snpEff

I did this: 
```
gemini load -t snpEff -p issue798.ped -v $vcf issue798.db
```
and had variant_impacts filled with difference transcripts for your example vcf.

**Comment by naumenko-sa** [comment]
Thanks Brent.
You are right - this vcf was annotated with snpEFF, it had ANN field rather than CSQ. Sorry, it was my fault. I was not able to share real samples with you, so I tried NA12878, which was processed with snpEFF.

Anyway, my VEP annotated vcf's contain just 1 effect per variant. But this is not because of GEMINI, I discovered that the pipeline uses --pick option of VEP
http://useast.ensembl.org/info/docs/tools/vep/script/vep_options.html,
which forces VEP to write exactly one effect per variant, and then GEMINI loads this into the database.

Thank you for you time spent on this issue, finally you helped me to resolve it.

Sergey
